### PR TITLE
[upnp] Controller only starts when Server is started

### PR DESCRIPTION
**Pull request by Rallymen007**
Follow up to #10249, hopefully I got it right this time.

### CODE REVIEW

**Comment by MartijnKaijser** [comment]
@Montellese

File: xbmc/network/NetworkServices.cpp [hunk 1]
```diff
@@ -829,12 +829,9 @@ bool CNetworkServices::StartUPnP()
...
-
-  if (CSettings::GetInstance().GetBool(CSettings::SETTING_SERVICES_UPNPCONTROLLER))
-  {
-    ret |= StartUPnPController();
+    ret |= StartUPnPServer(); 
+    if (CSettings::GetInstance().GetBool(CSettings::SETTING_SERVICES_UPNPCONTROLLER))
+      ret |= StartUPnPController();
   }
 
   if (CSettings::GetInstance().GetBool(CSettings::SETTING_SERVICES_UPNPRENDERER))
```
Comment (Montellese) [inline xbmc/network/NetworkServices.cpp]: Could you add a second commit to remove the curly braces belonging to this `if` statement?

**Comment by Montellese** [comment]
I just looked at the whole code again and wondering what the original problem was. `CNetworkServices::StartUPnP()` used to call all the different upnp-related `StartFoo()` methods. All those methods already contain logic to check if the respective settings are enabled i.e. `StartUPnPServer()` checks if `CSettings::SETTING_SERVICES_UPNPSERVER` is enabled. Why do we need to perform these checks in `CNetworkServices::StartUPnP()` as well?

File: xbmc/network/NetworkServices.cpp (hunk 1 above)
Comment (Rallymen007) [inline xbmc/network/NetworkServices.cpp]: Those two ifs are nested as to have the "controller needs server" behavior, and I think those curly braces are needed as the if block isn't one instruction long.

**Comment by Rallymen007** [comment]
Indeed they do some checks. If I recall correctly, my issue was that under no circumstances Kodi left port 1900 alone which is something I needed as the UPnP library I use has issues with port reuse despite all the flags being set properly.

This was the spirit of the original fix, but as you pointed out the only way to have this behavior is to fully recompile Kodi without UPnP as there is no suitable setting to shut down the UPnP client. The original fix helped me because of my wrong understanding of the settings, but after restoring the correct behavior it doesn't do shit.

So yeah, apart from adding a separate setting to shut down the UPnP client (which I could try but I'm sure I'll just butcher everything :p, not to mention it might be wrong) this fix is no longer useful.

**Comment by MartijnKaijser** [comment]
@Rallymen007 @Montellese should anything changed?

**Comment by Montellese** [comment]
@MartijnKaijser actually #10249 was unnecessary as well. We should probably revert it because all the checks that were added are already done in the UPnP logic.

**Pull request closed without merging**